include/realizations/catchment/Bmi_Multi_Formulation.hpp

```diff
@@ -245,6 +245,10 @@ namespace realization {
                 for(std::map<std::string,std::shared_ptr<data_access::GenericDataProvider>>::iterator iter = availableData.begin(); iter != availableData.end(); ++iter)
                 {
                     var_name = iter->first;
+                    auto it = std::find(forcing->get_available_variable_names().begin(), forcing->get_available_variable_names().end(), var_name);
+                    if (it == forcing->get_available_variable_names().end()) {
+                        var_name = forcing->get_available_variable_names()[0];
```

Comment:
The whole loop now boils down to this line. Just eliminate the `for` loop and set `var_name` to this... even the `try` block below is probably not useful now, because (I'm pretty sure) if `availableData[var_name]->get_data_start_time();` fails on a forcing var_name, something has gone wrong (in the current code, at least).

Actually... I think the whole use of a variable name is redundant in this code path (where `var_name == "*" || var_name = ""`... I think this whole section can be reduced to:

```
// Find one that successfully returns...
if(var_name == "*" || var_name == ""){
    return forcing->get_data_start_time();
}
// If not found ...
```

Can you try that?

Reply:
Actually the comment above that `if` is now not really correct either, probably just make it `// If no var_name, use forcing`... if the above works.

Reply:
Just tried. It seems to work.